### ISSUE TITLE
Accessibility: Button contrast

### ISSUE THREAD

**Issue by pauldambra** [PostHog/posthog#11214]
Not all of our button states pass color contrast checking (updated following #11223)

|           	| primary 	| danger 	| primary-alt 	 	| stealth 	|
|-----------	|---------	|--------		|-----------	|---------	|
| primary   	|         	| ❌      	|             	         	| ❌       	|
| secondary 	|         	|        	|             	         	|         	|
| tertiary  	|         	|        	|             	         	|         	|

## primary + danger

https://storybook.posthog.net/?path=/story/lemon-ui-lemon-button--default&args=type:primary;status:danger

`Element has insufficient color contrast of 4.26 (foreground color: #f4f6ff, background color: #db3707, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1`

## primary + stealth

https://storybook.posthog.net/?path=/story/lemon-ui-lemon-button--default&args=type:primary;status:stealth

`Element has insufficient color contrast of 1.07 (foreground color: #f4f6ff, background color: #ffffff, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1`

----------

Do the colours need to change? And/or are there combinations that shouldn't be allowed?

**Comment by pauldambra** [comment]
cc @benjackwhite @clarkus @lottiecoxon

**Comment by clarkus** [comment]
Love that you're thinking about and testing for this. We went through extensive contrast checks when updating the color palette recently and all the button colors passed contrast checks. There are button color combination in storybook now that don't actually align to our usage. There are no cases where we use a warning, success, or primary-alt button color combination in product. Those status colors are mostly used in banners, iconography, and tags to indicate status. 

List of button color combinations that aren't applicable to product at this time and should be removed from storybook as valid button combinations:
- https://storybook.posthog.net/?path=/story/lemon-ui-lemon-button--default&args=type:primary;status:success
- https://storybook.posthog.net/?path=/story/lemon-ui-lemon-button--default&args=type:primary;status:warning
- https://storybook.posthog.net/?path=/story/lemon-ui-lemon-button--default&args=type:secondary;status:warning
- https://storybook.posthog.net/?path=/story/lemon-ui-lemon-button--default&args=type:tertiary;status:warning
- https://storybook.posthog.net/?path=/story/lemon-ui-lemon-button--default&args=type:primary;status:muted-alt
- https://storybook.posthog.net/?path=/story/lemon-ui-lemon-button--default&args=type:tertiary;status:muted-alt
- https://storybook.posthog.net/?path=/story/lemon-ui-lemon-button--default&args=type:primary;status:stealth

[Here's the documentation](https://www.figma.com/file/Y9G24U4r04nEjIDGIEGuKI/PostHog-Design-System-One?node-id=3112%3A1254) on the current button states that we are using now. It's very much limited to primary, secondary, tertiary blues, and destructive primary, secondary, tertiary reds. The one combination that I found that didn't meet contrast goals is the active / hover state for destructive (red) buttons. 

![Button](https://user-images.githubusercontent.com/254612/183753080-2923bc0f-291b-4ef7-8794-33fc8cff9258.png)

**Comment by benjackwhite** [comment]
Nice idea Paul 👍 
I removed those statuses mentioned and got some typescript errors. Use cases I could find: 
**tertiary, muted-alt**
Insight Card button
<img width="154" alt="Screenshot 2022-08-10 at 08 14 53" src="https://user-images.githubusercontent.com/2536520/183828633-2d34c23b-bfe2-43ec-a8b8-8883efef37c8.png">
Input clear button
<img width="144" alt="Screenshot 2022-08-10 at 08 16 43" src="https://user-images.githubusercontent.com/2536520/183828944-4d62a4c7-ff79-413b-ac47-3029094fc4e8.png">

**primary, success**
End experiment button (couldn't get a screenshot as its a hard state to trigger)

----

@clarkus would love your advise on the Dashboard Insight "Show details" button as that's the only hard one to work around. I guess Experiment Success button could just become a standard primary button?

@pauldambra The input clear button is slightly unique in the Design System anyways in that it has dark text but is blue on hover (essentially `status=muted-alt` without hover and `status=primary` with. As such we can probably build that differently so it isn't using the muted-alt color (and then see if that is picked up by the accessibility bot ;) ).

So glad that the storybook surfaces all of this 😀 that was the motivation behind having all possible states visible (even if not commonly used) so we can see what _could_ end up happening in the UI.

**Comment by clarkus** [comment]
> @clarkus would love your advise on the Dashboard Insight "Show details" button as that's the only hard one to work around. I guess Experiment Success button could just become a standard primary button?

This work is from quite a while ago and predates some of the standardization we're doing now. Much like the insight query builder actions, these were intended to be smaller format so that they don't displace the other content. There was a ton of work put in to keep the card heights reasonable so you can see all the insight details while not unnecessarily displacing other insight cards further down the page. They could be updated to match the small button components, but I was trying to downplay their priority even more because they're not critical actions and I didn't want them to be read as higher priority than some other things in the cards. 

I think we should drop the `Secondary text · alternate` color palette option and instead use the secondary text color here. We don't really need both and the low contrast is a clear problem that's easily avoided.

**Comment by pauldambra** [comment]
So is that removing what the implementation calls `type=secondary status=muted-alt` and replacing it with `type=secondary status=muted-alt`

<img width="563" alt="Screenshot 2022-08-12 at 09 11 34" src="https://user-images.githubusercontent.com/984817/184313042-09d0a1d0-7897-40a5-a1b4-1df130978f6b.png">

Or removing _all_ `muted-alt` options in favour of `primary-alt`?

**Comment by pauldambra** [comment]
There are 40 usages of `text-muted-alt` CSS

But (I think) only two usages of button `muted-alt`

**Comment by clarkus** [comment]
> So is that removing what the implementation calls `type=secondary status=muted-alt` and replacing it with `type=secondary status=muted-alt`
> 
> <img alt="Screenshot 2022-08-12 at 09 11 34" width="563" src="https://user-images.githubusercontent.com/984817/184313042-09d0a1d0-7897-40a5-a1b4-1df130978f6b.png">
> 
> Or removing _all_ `muted-alt` options in favour of `primary-alt`?

The primary-alt and muted-alt buttons in those priorities don't exist that I know of. If they do, they could be swapped to secondary buttons. The exception is the tertiary action that uses the muted-alt option - these are sometimes used in components for low-priority utility actions (show legend, show filters, show details, etc). For contrast purposes, the primary-alt color palette option could be used in place of mued-alt for tertiary actions.

**Comment by pauldambra** [comment]
Down to just two now. Description updated...

Primary danger and primary stealth

**Comment by clarkus** [comment]
For contrast on primary destructive, it looks like the font-weight might be wrong. In figma the buttons use a mid-range emphasis value (with the Inter font, I'd say `font-weight:500` or so would possibly tip this over into the valid range. For reference, the contrast checker we use in figma rates this at 4.60:1.

<img width="241" alt="Screen Shot 2022-08-17 at 9 38 09 AM" src="https://user-images.githubusercontent.com/254612/185195167-3350c146-fcc5-412e-8185-66eba07acd89.png">
 
For the stealth option, I think it's objecting to the icon color. This is used muted-alt, but could be swapped to secondary text or primary-alt to improve contrast. I've changing usage in figma to use secondary text for new cases.

**Comment by pauldambra** [comment]
Ah... we have `foreground color: #f4f6ff` and with #fff it passes 🕵️‍♀️ 🔍
